qiskit/quantum_info/operators/operator.py

```diff
@@ -540,14 +540,17 @@ def compose(self, other: Operator, qargs: list | None = None, front: bool = Fals
         ret._op_shape = new_shape
         return ret
 
-    def power(self, n: float) -> Operator:
-        """Return the matrix power of the operator.
+    def power_if_unitary(self, exponent: float) -> Operator:
+        """Return the matrix power of the operator, provided that operator is
+        unitary.
+
+        This is faster than the ``power`` method.
```

Comment:
I think we might want to put a bigger warning in the docstring of this method that it'll produce nonsense if the operator wasn't unitary.

Also, super minor, but ` ``power`` ` can become `` :meth:`power` `` to insert a cross-ref.